### DISCUSSION TITLE
git with zsh-autosugesstions stuck with tab completion

### DISCUSSION THREAD

**Discussion by twmht** [ohmyzsh/ohmyzsh#12871] [Help]
When I use Git commands, such as:

```bash
git diff /path/to/
```

and press `Tab`, it freezes. My expected behavior is that if there are other files under `/path/to/`, I should be able to select them. How can I achieve this?

Here how i set up oh-my-zsh in my git bash

https://gist.github.com/glenkusuma/7d7df65a89e485ec2f4690fdc88fffd6#install-git-bash-with-zsh-oh-my-zsh--powerlevel10k--windows-guide

**Comment by roeniss** [comment]
Do you experience the same problem without any other plugins/settings?

**Comment by KianAnbarestani** [comment]
A common reason for “freezes” when tab-completing Git commands in Git Bash + Oh My Zsh is that Zsh is scanning a large directory tree for completions. It may not truly be frozen, just slow.

Try these steps:

1. Disable or limit Git completions temporarily:

```
# In ~/.zshrc
plugins=()  # Remove "git" if present
```
Then open a new shell and see if tab completion still lags.

2. Use a minimal Zsh session to confirm Oh My Zsh is the culprit:

```
zsh -f
# Then manually enable completion:
autoload -U compinit && compinit
```
Test `git diff /path/to/<TAB>` again. If it’s faster, something in your Oh My Zsh config or plugins is slowing it down.

3. Enable visual feedback:
```
# In ~/.zshrc
COMPLETION_WAITING_DOTS="true"
```
This shows dots while Zsh is generating completions, helping distinguish a real freeze from a time-consuming directory scan.

4. Narrow completion scope for Git: You can set Git completion to only show tracked files, ignoring untracked. For instance:
`zstyle ':completion:*:*:git:*' ignored-patterns '*~'`
or other patterns to skip large sets of files.

**If these changes resolve your issue, then you know it was large, slow Git completions in combination with your environment. If it still “freezes,” make sure Git Bash can handle forward slashes and that your Windows path translations aren’t interfering with completion.**
